### PR TITLE
Implementing Chaos

### PR DESCRIPTION
**Pull request by Mach131**
had a bit of fun with the buddy names

### CODE REVIEW

File: scenes/game/local_game.gd
Comment (daniel-k-taylor) [inline scenes/game/local_game.gd:482]: why bother putting Variant?

Comment (daniel-k-taylor) [reply scenes/game/local_game.gd:482]: i guess since it is assumed string maybe but this is nullable.

File: scenes/game/local_game.gd
Comment (daniel-k-taylor) [inline scenes/game/local_game.gd:1702]: I'm starting to worry about the amount of code we're duplicating between close/advance/retreat.
Maybe close/retreat can basically be the same function just in different directions?
And close is really just advance but opponent is blocking you.

I'll let you decide if you want to do this or not.

Comment (daniel-k-taylor) [reply scenes/game/local_game.gd:1702]: I know for Kokonoe I was also adding some stop_on_buddy functionality, which is slightly different from your blocked_by_buddy since it is inclusive.

File: data/card_definitions.json
Comment (daniel-k-taylor) [inline data/card_definitions.json:2314]: maybe instead of this, condition buddy_in_opponent_space used for litchi?

File: scenes/game/local_game.gd
Comment (daniel-k-taylor) [inline scenes/game/local_game.gd:2851]: shouldn't need any of this if you just use litchi's thing, buddy_in_opponent_space

File: data/card_definitions.json
Comment (daniel-k-taylor) [reply data/card_definitions.json:2314]: buddy shouldn't be in play unless they used their char ability

File: scenes/game/local_game.gd
Comment (daniel-k-taylor) [inline scenes/game/local_game.gd:3589]: the effect is greater_than, but this log says speed equal or greater.

File: scenes/game/local_game.gd
Comment (daniel-k-taylor) [inline scenes/game/local_game.gd:5163]: effect is named greater than, but this is >=

**Review by daniel-k-taylor — CHANGES_REQUESTED** [review]

File: scenes/game/game.gd
Comment (Mach131) [reply scenes/game/game.gd:495]: I didn't mean for the offset to be mirrored, if that's what you mean; the offset's direction seems consistent facing both sides

File: scenes/game/local_game.gd
Comment (Mach131) [reply scenes/game/local_game.gd:482]: yeah, I wanted to avoid making two variables for this; maybe that would be clearer?

File: data/card_definitions.json
Comment (Mach131) [reply data/card_definitions.json:2314]: will borrow the buddy_in_opponent_space condition; however, the buddy can also be in play because of the deep revenance boost

File: scenes/game/game.gd
Comment (daniel-k-taylor) [reply scenes/game/game.gd:495]: and it works for opponent's buddy too?
Just want to double check since there is opportunity for weirdness here

File: scenes/game/local_game.gd
Comment (daniel-k-taylor) [reply scenes/game/local_game.gd:482]: it's fine i guess

File: data/card_definitions.json
Comment (daniel-k-taylor) [reply data/card_definitions.json:2314]: oh weird, so you deep rev strike and if it's this it calculates range from yourself.
But it should still work in that case?

Comment (Mach131) [reply data/card_definitions.json:2314]: yup, it still gets the place in space effects; not sure how useful it is but

File: scenes/game/game.gd
Comment (Mach131) [reply scenes/game/game.gd:495]: yeah, looks the same on both clients

File: scenes/game/local_game.gd
Comment (Mach131) [reply scenes/game/local_game.gd:1702]: i wonder, is there anything stopping us from having advance/retreat/close just redirect to move_to (with a blocking option for close)?

Comment (daniel-k-taylor) [reply scenes/game/local_game.gd:1702]: Well, you don't know exactly where you're moving to depending on the opponent for advance.
I think the most reasonable is like:
move_in_direction_by_amount(direction : int, amount : int, blocked_by_opponent : bool)
Then it moves you in that direction 1 at a time, checking blocked_by_opponent, blocked_by_buddy_id stuff, and the new stop_on_buddy thing I'm adding.
When you land on the opponent (and you weren't blocked by them), you move a bonus space in direction.

How does that sound?

Comment (daniel-k-taylor) [reply scenes/game/local_game.gd:1702]: Close/Advance/Retreat will then calculate the direction first and pass that in.

Comment (Mach131) [reply scenes/game/local_game.gd:1702]: Probably more simple than what I was trying out; I'll see how well it works

File: scenes/game/local_game.gd
Comment (daniel-k-taylor) [inline scenes/game/local_game.gd:1641]: personally i would have just passed in direction as a 1 or -1 int, but this is less open to passing in weirdo stuff that is not 1 or -1.

File: scenes/game/local_game.gd
Comment (daniel-k-taylor) [inline scenes/game/local_game.gd:1665]: Shouldn't target_location then be changed to not be on the opponent?

Comment (daniel-k-taylor) [reply scenes/game/local_game.gd:1665]: oh, it breaks, got it

**Review by daniel-k-taylor — APPROVED** [review]
Love it.

**Pull request merged**
